include/picongpu/plugins/openPMD/WriteMeta.hpp

```diff
@@ -0,0 +1,256 @@
+/* Copyright 2013-2019 Axel Huebl, Franz Poeschel
+ *
+ * This file is part of PIConGPU.
+ *
+ * PIConGPU is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PIConGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+#include "picongpu/fields/FieldManipulator.hpp"
+#include "picongpu/fields/currentInterpolation/CurrentInterpolation.hpp"
+#include "picongpu/plugins/common/stringHelpers.hpp"
+#include "picongpu/plugins/openPMD/openPMDWriter.def"
+#include "picongpu/simulation_defines.hpp"
+#include "picongpu/traits/SIBaseUnits.hpp"
+
+#include <pmacc/Environment.hpp>
+
+#include <openPMD/openPMD.hpp>
+
+#include <list>
+#include <sstream>
+#include <string>
+
+
+namespace picongpu
+{
+namespace openPMD
+{
+    using namespace pmacc;
+
+    namespace writeMeta
+    {
+        /** write openPMD species meta data
+         *
+         * @tparam numSpecies count of defined species
+         */
+        template< uint32_t numSpecies =
+                      bmpl::size< VectorAllSpecies >::type::value >
```

Comment:
here and in other places, alignment is quite off...

Reply:
Is it better now? If yes, the fix was changing `AlignAfterOpenBracket: DontAlign` in [here](https://github.com/ax3l/contributing/blob/topic-clangFormat/IDESettings/.clang-format) to `AlignAfterOpenBracket: AlwaysBreak`.

Reply:
That `clang-format` file is unfortunately a WIP since not all our style rules are expressible in it yet.